logstreams/src/test/java/io/zeebe/logstreams/log/LogBlockIndexTest.java

```diff
@@ -231,6 +227,27 @@ public void shouldRecoverIndexFromSnapshot() throws Exception {
     assertThat(blockIndex.getLastPosition()).isEqualTo(snapshotPosition);
   }
 
+  @Test
+  public void shouldDeleteEntriesUpToPosition() {
+    // given
+    final int numBlocks = 3;
+    addBlocks(numBlocks);
+
+    // when
+    blockIndex.deleteUpToPosition(indexContext, ENTRY_OFFSET);
+
+    // then
+    assertThat(blockIndex.lookupBlockAddress(indexContext, 0))
+        .isEqualTo(LogBlockIndex.VALUE_NOT_FOUND);
+    for (int i = 0; i < ENTRY_OFFSET; ++i) {
+      assertThat(blockIndex.lookupBlockPosition(indexContext, i))
+          .isEqualTo(LogBlockIndex.VALUE_NOT_FOUND);
+    }
+    assertThat(blockIndex.lookupBlockAddress(indexContext, ENTRY_OFFSET))
```

Comment:
I think the address has to be `ENTRY_OFFSET * ADDRESS_MULTIPLIER` or?

Reply:
Can we check for the blockposition as well?

Reply:
True, I'll change that